src/subscriber.rs

```diff
@@ -216,6 +218,23 @@ impl Iterator for Subscriber {
     }
 }
 
+#[doc(hidden)]
+#[must_use = "futures do nothing unless you `.await` or poll them"]
+struct Next<'a> {
+    subscriber: &'a mut Subscriber,
+}
+
+impl<'a> Future for Next<'a> {
+    type Output = Option<Event>;
+
+    fn poll(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+    ) -> Poll<Self::Output> {
+        Pin::new(&mut *self.subscriber).poll_next(cx)
```

Comment:
`self.subscriber` is an `&mut Subscriber` but it can't move, so `&mut *self.subscriber` to make it move-able.